packages/java-edition/src/binder/index.ts

```diff
@@ -166,17 +169,50 @@ export function* getRels(
 	return undefined
 }
 
-export function dissectUri(uri: string, ctx: ContextBase) {
+export function dissectUri(uri: string, ctx: UriBinderContext) {
 	const regex =
 		/^data\/([^\/]+)\/((?:tags\/|worldgen\/)?[a-z_]+)\/(.*)(\.(?:mcfunction|json))$/
 
 	const rels = getRels(uri, ctx.roots)
+
+	const { customResources } = ctx.config.env
+
+	if (
+		customResources.length > 0 &&
+		ctx.project['customResourcesDefined'] === undefined
+	) {
+		ctx.project['customResourcesDefined'] = 'done'
```

Comment:
What's the reason for this state? This feels highly error prone.

Reply:
Its an optimization measure, if you can figure out a better way to handle this please let me know :)

Reply:
Okay I see. I don't think this is necessary, especially since tha large majority of people won't even use this config